posthog/hogql/test/test_printer.py

```diff
@@ -60,16 +60,29 @@ def test_fields_and_properties(self):
             "replaceRegexpAll(JSONExtractRaw(properties, %(hogql_val_0)s), '^\"|\"$', '')",
         )
 
-        context = HogQLContext()
+        context = HogQLContext(within_non_hogql_query=True, using_person_on_events=False)
         self.assertEqual(
             self._expr("person.properties.bla", context),
-            "replaceRegexpAll(JSONExtractRaw(events__pdi__person.properties, %(hogql_val_0)s), '^\"|\"$', '')",
+            "replaceRegexpAll(JSONExtractRaw(person_props, %(hogql_val_0)s), '^\"|\"$', '')",
         )
 
-        context = HogQLContext(within_non_hogql_query=True, using_person_on_events=False)
+        context = HogQLContext(within_non_hogql_query=True, using_person_on_events=True)
         self.assertEqual(
             self._expr("person.properties.bla", context),
-            "replaceRegexpAll(JSONExtractRaw(person_props, %(hogql_val_0)s), '^\"|\"$', '')",
+            "replaceRegexpAll(JSONExtractRaw(person_properties, %(hogql_val_0)s), '^\"|\"$', '')",
+        )
+
+        context = HogQLContext(within_non_hogql_query=False, using_person_on_events=False)
+        self.assertEqual(
+            self._expr("person.properties.bla", context),
+            "events__pdi__person.properties___bla",
+        )
+
+        context = HogQLContext(within_non_hogql_query=False, using_person_on_events=True)
+        self.assertEqual(
+            # TODO: for now, explicitly writing "poe." to opt in. Automatic switching will come soon.
```

Comment:
HogQL is not yet able to turn PoE (in its current form) on or off depending on the team's settings. It is doing this inside existing insights (non-hogql queries), but not in full hogql queries. This is planned for another PR.